test/select.jl

```diff
@@ -2346,6 +2349,180 @@ end
+    @test df.x1 !== df.x3
+    @test df.x2 !== df.x3
+    @test df.x1 === x
+
+    df = DataFrame(a=1:3)
+    x = df.a
+    transform!(df, :a .=> [:b])
+    @test df.a !== df.b
+    @test df.a === x
+
+    df = DataFrame(a=1:3)
+    x = df.a
+    transform!(df, :a .=> [:x1, :x2, :x3])
+    @test df.a !== df.x1
+    @test df.a !== df.x2
+    @test df.a !== df.x3
+    @test df.x1 !== df.x2
+    @test df.x1 !== df.x3
+    @test df.x2 !== df.x3
+    @test df.a === x
+
+    for cc in (true, false)
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a => :b, :a => :c, copycols=cc)
+        @test df2.b !== df2.c
+        @test (df2.b === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a, :a => :c, copycols=cc)
+        @test df2.a !== df2.c
+        @test (df2.a === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a => :c, :a, copycols=cc)
+        @test df2.a !== df2.c
+        @test (df2.c === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a => :c, :, copycols=cc)
+        @test df2.a !== df2.c
+        @test (df2.c === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a .=> [:c], :, copycols=cc)
+        @test df2.a !== df2.c
+        @test (df2.c === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a .=> [:c], :, :a => :b, copycols=cc)
+        @test df2.a !== df2.c
+        @test df2.b !== df2.c
+        @test df2.a !== df2.b
+        @test (df2.c === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = select(df, :a .=> [:x1, :x2, :x3], copycols=cc)
+        @test df2.x1 !== df2.x2
+        @test df2.x1 !== df2.x3
+        @test df2.x2 !== df2.x3
+        @test (df2.x1 === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = transform(df, :a .=> [:b], copycols=cc)
+        @test df2.a !== df2.b
+        @test (df2.a === x) ⊻ cc
+
+        df = DataFrame(a=1:3)
+        x = df.a
+        df2 = transform(df, :a .=> [:x1, :x2, :x3], copycols=cc)
+        @test df2.a !== df2.x1
+        @test df2.a !== df2.x2
+        @test df2.a !== df2.x3
+        @test df2.x1 !== df2.x2
+        @test df2.x1 !== df2.x3
+        @test df2.x2 !== df2.x3
+        @test (df2.a === x) ⊻ cc
+    end
+
+    # super rare cases of multicolumn output
+    # we do a proper detection only if source was a single column
+    df = DataFrame(a=1:3)
+    df2 = select(df, :a => (x -> (b=x, c=x)) => AsTable, copycols=false)
+    @test df.a === df2.b
+    @test df.a !== df2.c
+    df = DataFrame(a=1:3)
+    df2 = select(df, :a => (x -> DataFrame(b=x, c=x, copycols=false)) => AsTable, copycols=false)
+    @test df.a === df2.b
+    @test df.a !== df2.c
+    df = DataFrame(a=1:3)
+    df2 = transform(df, :a => (x -> (b=x, c=x)) => AsTable, copycols=false)
+    @test df.a === df2.a
+    @test df.a !== df2.b
+    @test df.a !== df2.c
```

Comment:
Check that `df2.b !== df2.c`? Same below.

Reply:
added